### ISSUE TITLE
Cleanup Lintian Warnings and Errors

### ISSUE THREAD

**Issue by lweberk** [SOCI/soci-pkg#10]
Currently lintian has a lot to say about things done wrong in debian packaging.

**Comment by OniDaito** [comment]
I shall take a look at that today. cheers

**Comment by OniDaito** [comment]
Currently I have the following warnings and errors:

    W: libsoci-core: package-name-doesnt-match-sonames libsoci-core4.0
    W: libsoci-core: extra-license-file usr/share/doc/libsoci-core/LICENSE_1_0.txt
    W: libsoci-core: non-dev-pkg-with-shlib-symlink usr/lib/libsoci_core.so.4.0.0 usr/lib/libsoci_core.so
    W: libsoci-mysql: package-name-doesnt-match-sonames libsoci-mysql4.0
    W: libsoci-mysql: non-dev-pkg-with-shlib-symlink usr/lib/libsoci_mysql.so.4.0.0 usr/lib/libsoci_mysql.so
    W: libsoci-postgresql: package-name-doesnt-match-sonames libsoci-postgresql4.0
    W: libsoci-postgresql: non-dev-pkg-with-shlib-symlink usr/lib/libsoci_postgresql.so.4.0.0 usr/lib/libsoci_postgresql.so
    E: libsoci-doc: privacy-breach-logo usr/share/doc/libsoci-doc/www/articles.html
    E: libsoci-doc: privacy-breach-logo usr/share/doc/libsoci-doc/www/doc.html 
    E: libsoci-doc: privacy-breach-logo usr/share/doc/libsoci-doc/www/doc/index.html
    E: libsoci-doc: privacy-breach-logo usr/share/doc/libsoci-doc/www/events.html
    E: libsoci-doc: privacy-breach-logo usr/share/doc/libsoci-doc/www/index.html
    E: libsoci-doc: privacy-breach-logo usr/share/doc/libsoci-doc/www/links.html
    E: libsoci-doc: privacy-breach-logo usr/share/doc/libsoci-doc/www/people.html
    W: libsoci-odbc: package-name-doesnt-match-sonames libsoci-odbc4.0
    W: libsoci-odbc: non-dev-pkg-with-shlib-symlink usr/lib/libsoci_odbc.so.4.0.0 usr/lib/libsoci_odbc.so
    W: libsoci-sqlite3: package-name-doesnt-match-sonames libsoci-sqlite3-4.0
    W: libsoci-sqlite3: non-dev-pkg-with-shlib-symlink usr/lib/libsoci_sqlite3.so.4.0.0 usr/lib/libsoci_sqlite3.so

The errors are all relating to the documentation which is something currently being discussed in the main soci project. The remaining warnings I'm less worried about

**Comment by lweberk** [comment]
True, but I doubt the package making it into the official debian repositories in the current state.

For the description of the lintian error: https://lintian.debian.org/tags/privacy-breach-logo.html . It basicaly sais, to swap external references for static includes of them. Which makes sense.

As for the warnings:
package-name-doesnt-match-sonames: I consider this one relevant, as soci runs multiple SONAME versions at the same time, the current master is not (? not verified ?) compatible with release/3.2 which is a officialy release version of the library. Thus if debian stable/testing/unstable/experimental are to be able to run in parallel with differing SONAMEs, it would be sensible to adapt naming accordingly to avoid package collisions.
non-dev-pkg-with-shlib-symlink: Probably overridable for the time being. Would have to look deeper into it to really get the idea of how it is supposed to be handled.
